lib/Index/Index.cpp

```diff
@@ -856,6 +856,8 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
 
     if (Data.isImplicit)
       Info.roles |= (unsigned)SymbolRole::Implicit;
+    if (Data.Kind == SemaReferenceKind::DynamicMemberRef)
+      Info.roles |= (unsigned)SymbolRole::Dynamic;
```

Comment:
It'd be great if this could be handled with `reportIsDynamicRef` instead so that we get the receiver type relations.

Reply:
we worked on this offline and decided to punt on it since it also marks standard class initializers as dynamic. I will submit a follow up change as an example of how it could work